spec/std/http/formdata/generator_spec.cr

```diff
@@ -0,0 +1,76 @@
+require "http"
+require "spec"
+
+describe HTTP::FormData::Generator do
+  it "generates valid form-data messages" do
+    io = MemoryIO.new
+    HTTP::FormData.generate(io, "fixed-boundary") do |g|
+      g.field("foo", "bar")
+      g.field("baz", "qux", HTTP::Headers{"X-Testing" => "headers"})
+
+      body = MemoryIO.new "file content"
+      time = Time.new(2016, 1, 1, 12, 0, 0, kind: Time::Kind::Utc)
+      metadata = HTTP::FormData::FileMetadata.new("filename.txt \"", time, time, time, 12_u64)
+      headers = HTTP::Headers{"Foo" => "Bar", "Baz" => "Qux"}
+      g.file("file-test", body, metadata, headers)
```

Comment:
I wouldn't block this PR on it, but in the future we probably should have convenience API in `Generator` that takes the fields of `FileMetadata` as keyword arguments and a plain hash as headers I guess.